docs/explanations.md

```diff
@@ -0,0 +1,212 @@
+## nyaml Workflow
+For a given input file, the `nyaml` converter checks for the correct file type and call appropriate converter. For an XML file, the `nxdl2yaml` converter parses the `XML` file, by means of [lxml](https://lxml.de/) python library, into an `XML` tree object. Adhering to the NXDL rules, the converter writes the application definition or the base class object to a `yaml` file that matches the `nyaml` syntax. If the input file is a `yaml` file, the `yaml2nxdl` converter collects the comments in a `Comments` object and parses the `yaml` file into a python `dictionary` object. Later, the application definition or base classes will be converted into an `XML` file by combining the `Comments` and the python `dictionary` object.
```

Comment:
"Provided an input file, the `nyaml` converter ..... and calls one of two possible converters: For an `nxdl.xml` file this is the `nxdl2yaml` converter. For a `yaml`file this is the `yaml2nxdl` converter.

for the sentence with lxml reformulate "The lxml Python library (add proper ref) is used for parsing XML files into an `XML` tree object in memory."

Reply:
Tailored!